CMakeLists.txt

```diff
@@ -155,6 +155,18 @@ install(
     PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ
 )
 
+#
+# Install targets
+#
+
+install(EXPORT uvwConfig NAMESPACE uvw:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/uvw)
+install(TARGETS uvw EXPORT uvwConfig ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+if(FETCH_LIBUV AND BUILD_UVW_LIBS)
```

Comment:
I'm not fully convinced by this installation of `uv_a` and `uv`.
Is our responsability to install libuv on the HOST system?

Reply:
This was the previous behavior in uvw so I just left it asis:
see https://github.com/skypjack/uvw/blob/3d60ab1a6d98d53c1180a7ca793f5dc7ce1290f6/src/CMakeLists.txt#L88

Reply:
Yes it's rare that someone FETCH_LIBUV and then installs uvw at the same time.

for me the installation step only makes sense when packaging uvw
in this case, it shouldn't use libuv from our side (the distribution package would be more aappropriate)

Reply:
> This was the current behavior in uvw so I just left it asis

Please, don't trust us and feel free to change it if you master `cmake` better than what we do (that is likely the case, at least for me 😅).
I'm open to listen to your suggestions to improve it overall, don't care about how it was.

Reply:
Unfortunately, after a bunch of tests, it seems that: 
- we can't remove this installation of libuv when uvw is built as a library (that is, uvw is linked with either `uv::uv-share` or `uv::uv-static`), CMake complains that 

```
CMake Error: install(EXPORT "uvwConfig" ...) includes target "uvw" which requires target "uv_a" that is not in any export set.
```

```
CMake Error: install(EXPORT "uvwConfig" ...) includes target "uvw" which requires target "uv" that is not in any export set.
```

in my point of view, this means when something is being got "exported", all its dependencies must also be exported.

Reply:
Yes, and again I think it's really rare that someone fetches libuv using `FETCH_LIBUV` and then "install" uvw.
There's no way to prevent a user from doing "`cmake --install .`", should we print an "unsuggested" warning message when someone is doing it in this unclean way?

Reply:
> when building as a library, the install command causes libuv to be installed, this is truly unwanted behavior

one specific version of `uvw` is not compatible with many versions of libuv, but the specific version it has been developed upon. So I think that it's very rare that the OS offers the right version of libuv of the version of uvw you need to install. In summary, it's correct to install uv_a and uv as targets under the same path of uvw-static and uvw-shared, so the linkage should go well 11 9's of time.. do you guys agree on this?

Reply:
> In my point, I'd suggest providing a "USE_SYSTEM_LIBUV:BOOL" option for long term ease of use.

This can be a way, but I think to remember we avoided this way because of the reason explained before (OSes never give you the right libuv version you need).

Reply:
Oh yes, this makes sense, if that's the case, libuv should be installed into the host system then.

Reply:
From reviewing the discussion, I cannot understand if the `libuv` installation will overwirte the existing one in the HOST or will be under something related to `uvw`. Anyway I will double check this on my own and come back to you.

Reply:
Ok I double checked this: as it currently is (and 99% as it probably was) the `make install` overwrites the existing libuv on the system.
IMO that's really not desirable.. @skypjack and @moodyhunter do you have any suggestion to prevent this?

Reply:
I have zero knowledge on solving this :(
I guess a possible way could be installing libuv to somewhere else (e.g. include/uvw/bundledUV/)?

Reply:
Far from being a cmake expert but isn't it what cmake `install`'s `LIBRARY DESTINATION` or similar is meant for?

Reply:
Yes, the CMake's `SOMETHING DESTINATION` controls where the installed files will be.
More information can be found here: https://cmake.org/cmake/help/latest/command/install.html#targets

Reply:
I agree with both of you, but the target libuv is not defined by us, but by the mainstream recipe :(
Maybe we have a way to override the installation behavior. Looking into it later today.

Reply:
Ok, I got the idea from a previous project of mine. Here is a sample commit that shows the outcomes of `sudo make install`.
https://github.com/stefanofiorentino/uvw/tree/moodyhunter-246

```
-- Install configuration: "Debug"
-- Up-to-date: /usr/local/include/uvw/async.h
-- Up-to-date: /usr/local/include/uvw/check.h
-- Up-to-date: /usr/local/include/uvw/config.h
-- Up-to-date: /usr/local/include/uvw/dns.h
-- Up-to-date: /usr/local/include/uvw/emitter.h
-- Up-to-date: /usr/local/include/uvw/fs.h
-- Up-to-date: /usr/local/include/uvw/fs_event.h
-- Up-to-date: /usr/local/include/uvw/fs_poll.h
-- Up-to-date: /usr/local/include/uvw/handle.hpp
-- Up-to-date: /usr/local/include/uvw/idle.h
-- Up-to-date: /usr/local/include/uvw/lib.h
-- Up-to-date: /usr/local/include/uvw/loop.h
-- Up-to-date: /usr/local/include/uvw/pipe.h
-- Up-to-date: /usr/local/include/uvw/poll.h
-- Up-to-date: /usr/local/include/uvw/prepare.h
-- Up-to-date: /usr/local/include/uvw/process.h
-- Up-to-date: /usr/local/include/uvw/request.hpp
-- Up-to-date: /usr/local/include/uvw/resource.hpp
-- Up-to-date: /usr/local/include/uvw/signal.h
-- Up-to-date: /usr/local/include/uvw/stream.h
-- Up-to-date: /usr/local/include/uvw/tcp.h
-- Up-to-date: /usr/local/include/uvw/thread.h
-- Up-to-date: /usr/local/include/uvw/timer.h
-- Up-to-date: /usr/local/include/uvw/tty.h
-- Up-to-date: /usr/local/include/uvw/type_info.hpp
-- Up-to-date: /usr/local/include/uvw/udp.h
-- Up-to-date: /usr/local/include/uvw/underlying_type.hpp
-- Up-to-date: /usr/local/include/uvw/util.h
-- Up-to-date: /usr/local/include/uvw/work.h
-- Up-to-date: /usr/local/include/uvw.hpp
-- Up-to-date: /usr/local/lib/cmake/uvw/uvwConfig.cmake
-- Up-to-date: /usr/local/lib/cmake/uvw/uvwConfig-debug.cmake
-- Up-to-date: /usr/local/lib/libuvw.a
-- Up-to-date: /usr/local/include/uvw/uv/include
-- Up-to-date: /usr/local/include/uvw/uv/include/uv.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/errno.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/linux.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/android-ifaddrs.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/stdint-msvc2008.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/unix.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/os390.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/win.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/sunos.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/darwin.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/tree.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/version.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/posix.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/aix.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/bsd.h
-- Up-to-date: /usr/local/include/uvw/uv/include/uv/threadpool.h
-- Up-to-date: /usr/local/lib/uvw/libuv_a.a
-- Up-to-date: /usr/local/lib/uvw/libuv.so.1.0.0
-- Up-to-date: /usr/local/lib/uvw/libuv.so.1
-- Up-to-date: /usr/local/lib/uvw/libuv.so
-- Up-to-date: /usr/local/lib/pkgconfig/libuvw-static.pc
```
I worry that also pkg-config has to be changed (or even better removed).

Reply:
So, that's fine, it installs libuv under uvw as it ought to be, isn't it? I vote for getting rid of pkg-config btw. 😅

Reply:
I don't use pkg-config so can't provide any constructive suggestions.